src/Castle.Facilities.AspNetCore/Exceptions/LifestyleNotSupportedException.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright 2004-2018 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Facilities.AspNetCore.Exceptions
+{
+	using System;
+
+	using Castle.Core;
+
+	/// <summary>
+	/// For when an unsupported <see cref="LifestyleType"/> get passed to the <see cref="CastleWindsorRegistrationExtensions.CrossWired{TService}(Castle.MicroKernel.Registration.ComponentRegistration{TService},Castle.Windsor.IWindsorContainer,Microsoft.Extensions.DependencyInjection.IServiceCollection,Castle.Core.LifestyleType)"/> extension method.
+	/// </summary>
+	public class LifestyleNotSupportedException : Exception
```

Comment:
Is a custom exception really necessary for a situation no one is going to specifically catch, I thought you were just going to use a `NotSupportedException` or even better a `ArgumentException`.

Reply:
Thought a custom exception with a namespace and a custom message would make the stack trace a little cleaner. Will change it back.